runtime/src/snapshot_package.rs

```diff
@@ -42,34 +41,61 @@ pub struct AccountsPackage {
     pub block_height: Slot,
     pub slot_deltas: Vec<BankSlotDelta>,
     pub snapshot_links: TempDir,
-    pub storages: SnapshotStorages,
+    pub snapshot_storages: SnapshotStorages,
     pub hash: Hash, // temporarily here while we still have to calculate hash before serializing bank
     pub archive_format: ArchiveFormat,
     pub snapshot_version: SnapshotVersion,
     pub snapshot_archives_dir: PathBuf,
     pub expected_capitalization: u64,
     pub hash_for_testing: Option<Hash>,
     pub cluster_type: ClusterType,
+    pub snapshot_type: Option<SnapshotType>,
 }
 
 impl AccountsPackage {
-    /// Create an accounts package
+    /// Package up bank files, storages, and slot deltas for a snapshot
     #[allow(clippy::too_many_arguments)]
-    fn new(
+    pub fn new(
         bank: &Bank,
         bank_snapshot_info: &BankSnapshotInfo,
-        status_cache_slot_deltas: Vec<BankSlotDelta>,
+        bank_snapshots_dir: impl AsRef<Path>,
+        slot_deltas: Vec<BankSlotDelta>,
         snapshot_archives_dir: impl AsRef<Path>,
         snapshot_storages: SnapshotStorages,
         archive_format: ArchiveFormat,
         snapshot_version: SnapshotVersion,
         hash_for_testing: Option<Hash>,
-        snapshot_tmpdir: TempDir,
+        snapshot_type: Option<SnapshotType>,
     ) -> Result<Self> {
+        info!(
+            "Package snapshot for bank {} has {} account storage entries (snapshot type: {:?})",
+            bank.slot(),
+            snapshot_storages.len(),
+            snapshot_type,
+        );
+
+        if let Some(SnapshotType::IncrementalSnapshot(incremental_snapshot_base_slot)) =
+            snapshot_type
+        {
+            assert!(
+                bank.slot() > incremental_snapshot_base_slot,
+                "Incremental snapshot base slot must be less than the bank being snapshotted!"
+            );
+            assert!(
+            snapshot_storages.iter().all(|storage| storage
+                .iter()
+                .all(|entry| entry.slot() > incremental_snapshot_base_slot)),
+            "Incremental snapshot package must only contain storage entries where slot > incremental snapshot base slot (i.e. full snapshot slot)!"
+            );
+        }
+
         // Hard link the snapshot into a tmpdir, to ensure its not removed prior to packaging.
+        let snapshot_links = tempfile::Builder::new()
+            .prefix(&format!("{}{}-", TMP_BANK_SNAPSHOT_PREFIX, bank.slot()))
```

Comment:
Do full snapshot and incremental snapshot look any different now for the tmp dir entry?

Reply:
Nope. None of the different temp directories need different prefixes actually, but now we're only differentiating between bank snapshots and snapshot archives.